test/plural_resolver_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:i18next/i18next.dart';
+import 'package:i18next/src/plural_resolver.dart';
+import 'package:mockito/mockito.dart';
+
+abstract class FakeCheckerKeyFunction {
+  bool call(String pluralModifier);
+}
+
+class MockCheckerKeyFunction extends Mock implements FakeCheckerKeyFunction {}
+
+void main() {
+  group('PluralResolver', () {
+    MockCheckerKeyFunction checkerKeyFnc;
+    PluralResolver subject;
+
+    setUpAll(() {
+      checkerKeyFnc = MockCheckerKeyFunction();
+      subject = PluralResolver();
+    });
+
+    test('when count is equal 1 returns empty string', () {
+      const count = 1;
+      final result = subject.pluralize(
+        count,
+        I18NextOptions.base,
+        checkerKeyFnc,
+      );
+
+      verifyZeroInteractions(checkerKeyFnc);
+      expect(result, equals(''));
```

Comment:
Instead of the mock, you could use the `expectAsync1` instead.

Reply:
What do u think to leave this to an 'integration' tests?

Reply:
Integration tests don't have direct access to the function parameter.

The suggestion is only to remove the usage of a mocking class altogether:
So for this test, it would be:

```dart
...
final result = subject.pluralize(..., expectAsync1(null, count: 0));
expect(result, equals(''));
```

Reply:
I'm only suggesting it because of the repeated pattern of mocking a method with a single value then expecting if it was called or not (which is what expectAsync is for)